Add documentation for building and publishing a WooCommerce Core Release

prettyboymp opened the pull request:
### Changes proposed in this Pull Request:

This adds a new section to the releases documentation to cover the build and publish process for releasing a new WooCommerce version.

Closes WOOPLUG-4567, #58704 .

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [X] This Pull Request does not require a changelog entry. (Comment required below)
Documentation only.
<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

prettyboymp committed:
Add documentation for building and publishing a WooCommerce Core Release

prettyboymp committed:
Add point out to building and publishing in main content.
Removing pending note for some of the workflow

prettyboymp committed:
Add link to internal staging doc

jorgeatorres commented on docs/contribution/releases/building-and-publishing.md:
```diff
@@ -0,0 +1,48 @@
+3. Validate that the new release is available:
+   * https://plugins.svn.wordpress.org/woocommerce/tags/ should list the version number.
+   * The new version should also be available inside the Previous versions dropdown [inside the Advanced Options screen on the WordPress.org plugin page](https://wordpress.org/plugins/woocommerce/advanced/).
+4. If this is not an `RC` release, [Deploy the release to our staging environments](https://wp.me/PCYsg-18BQ) and monitor for any issues.
```
Should we add a step to bump the stable tag using [this workflow](https://github.com/woocommerce/woocommerce/actions/workflows/release-update-stable-tag.yml) as well?

jorgeatorres commented on docs/contribution/releases/building-and-publishing.md:
```diff
@@ -0,0 +1,48 @@
+
+## Prerequisites
+
+* You should have a WordPress.org account with committer access to the WooCommerce plugin, as well as the associated SVN credentials.
```
While this is necessary when performing the steps 'manually', with the new workflows it shouldn't be. The repo should already be configured with the relevant secrets.

prettyboymp replied:
@jorgeatorres, while the release lead doesn't need SVN credentials, they do still need committer access to approve the release to WP.org, correct?

jorgeatorres replied:
Correct, though I guess we want people to start using one of the Woo accounts, otherwise everyone will have to have commit access to the repo. Not sure if we should mention that here.

Guess we can leave this as is, though the SVN creds in particular won't be needed.

jorgeatorres commented on docs/contribution/releases/building-and-publishing.md:
```diff
@@ -0,0 +1,48 @@
+4. If this is not an `RC` release, [Deploy the release to our staging environments](https://wp.me/PCYsg-18BQ) and monitor for any issues.
+5. Publish the previously created GitHub [draft release tag](https://github.com/woocommerce/woocommerce/releases).
+   * Check off Set as a pre-release if you’re releasing an RC. Otherwise, check off Set as the latest release.
+6. Ping Dev Advocacy in the #woo-core-releases channel in Slack notifying them of the release and asking them to update or publish the release post. Provide links to both the GitHub tag as well as the ZIP from WordPress.org as part of your message.
```
I believe this is no longer necessary. Can you confirm @albarin?

prettyboymp replied:
Should running the [`Release: Post-release published flows`](https://github.com/woocommerce/woocommerce/blame/trunk/.github/workflows/release-new-release-published.yml) replace this step?

jorgeatorres replied:
You mean step 6? I don't think it's that particular workflow but we do have some notifications going to Slack when a new release is marked as stable. I might be missing something, so I'll let @albarin confirm.

albarin replied:
We ask DevAdv to publish the release post with this https://github.com/woocommerce/woocommerce/blob/7fceed1befb9f456225cfed20298a1160075d666/.github/workflows/release-new-release-published.yml#L1-L22 and then we notify again when it's marked as stable in https://github.com/woocommerce/woocommerce/blob/7fceed1befb9f456225cfed20298a1160075d666/.github/workflows/release-update-stable-tag.yml#L235-L247

prettyboymp replied:
Ok.  I removed the final two steps that mentioned sending slack notifications since they're covered.  I don't think it's a big deal, but the woo announcement slack ping may go out before the notice to DevAdv.  

One thing I think we're missing is updating the github release tag with a link to the release announcement.  But I think we should update that later when we go back through and clean things up.

jorgeatorres reviewed (commented):
Hey @prettyboymp!

Thanks for working on this and for incorporating some of the new workflows! 💯

Overall, the instructions look good to me. A few steps might be simplified once we review / merge a couple other workflows that we're still working on, but if this PR gets merged first, we can update the docs later, which is fine.

I left some feedback, but nothing serious. Let me know what you think.

prettyboymp committed:
Add the release-update-stable step

prettyboymp committed:
Removing notification steps as they're already included within current workflows

prettyboymp committed:
Merge branch 'trunk' into WOOPLUG-4567-document-general-release-workflow

prettyboymp committed:
Removing direct reference to staging in light of https://a8c.slack.com/archives/C02NVV8LQLW/p1750673261093809 so we can keep this more focused on the view an outside contributor would see.
We will reference the documentation for the staging portion in the p2 post kicking off the release internally.

jorgeatorres reviewed (approved):
LGTM. Thank you @prettyboymp!

prettyboymp committed:
Merge branch 'trunk' into WOOPLUG-4567-document-general-release-workflow
